lib/Sema/ConstraintSystem.cpp

```diff
@@ -2794,6 +2794,11 @@ void ConstraintSystem::resolveOverload(ConstraintLocator *locator,
       choice.getDecl()->getAttrs().hasAttribute<DisfavoredOverloadAttr>()) {
     increaseScore(SK_DisfavoredOverload);
   }
+
+  if (choice.getKind() == OverloadChoiceKind::DeclViaUnwrappedOptional &&
```

Comment:
@Jumhyn Does this affects situations like https://bugs.swift.org/browse/SR-11535 in some way?

Reply:
It would be good to add a test-case from that SR to the suite.

Reply:
I don't think so, in that situation the `Optional.none` overload doesn't provide a valid solution since the generic argument to `Foo` cannot be inferred, and then we find the `Foo.none` solution when we attempt to bind the result of the `UnresolvedMemberExpr` to `Foo` (rather than `Foo?`). I'll verify that the behavior is the same, though.

Reply:
Yeah, seems like you are correct! I didn't note that. So it means this SR isn't really a bug...

Reply:
But maybe we can add this specific case to the suit and resolve it? WDYT? I can do this in a separated PR no problem :)

Reply:
I’m happy to add it as part of this PR, but probably won’t have a chance to make edits until tomorrow evening at the earliest. If you want to put up a PR before then, feel free!

Reply:
It's there #34744 
Actually, I found that we already have the test case added, so I just adjusted and added an explanation comment :)